src/server/replica.cc

```diff
@@ -438,33 +442,63 @@ error_code Replica::InitiateDflySync() {
   // Blocked on until all flows got full sync cut.
   fibers_ext::BlockingCounter sync_block{num_df_flows_};
 
+  // Atomically switch to new error handler.
   auto err_handler = [this, sync_block](const auto& ge) mutable {
     sync_block.Cancel();      // Unblock this function.
     DefaultErrorHandler(ge);  // Close sockets to unblock flows.
+    service_.SwitchState(GlobalState::LOADING, GlobalState::ACTIVE);  // Reset global state.
```

Comment:
I think that the error handler should make sure that all the flows are unblocked, but not do the cleanup
So in this case I would create an absl::Cleanup which will 
JoinAllFlows
switch the state 
and switch the error handler